qa/tests/sc_generate_test_data.py

```diff
@@ -0,0 +1,285 @@
+        we0_end_mcblock_hash = mc_node.generate(8)[7]
+        scblock_id2 = generate_next_blocks(sc_node, "first node", 3)[2]
+        check_mcreferencedata_presence(we0_end_mcblock_hash, scblock_id2, sc_node)
+
+        # Generate first mc block of the next epoch
+        we1_1_mcblock_hash = mc_node.generate(1)[0]
+        print("End mc block hash in withdrawal epoch 0 = " + we0_end_mcblock_hash)
+        scblock_id3 = generate_next_blocks(sc_node, "first node", 1)[0]
+        check_mcreference_presence(we1_1_mcblock_hash, scblock_id3, sc_node)
+
+        # Wait until Certificate will appear in MC node mempool
+        attempts = 20
+        while mc_node.getmempoolinfo()["size"] == 0 and attempts > 0:
+            print("Wait for certificate in mc mempool...")
+            time.sleep(10)
+            attempts -= 1
+            sc_node.block_best() # just a ping to SC node. For some reason, STF can't request SC node API after a while idle.
+        assert_equal(1, mc_node.getmempoolinfo()["size"], "Certificate was not added to Mc node mmepool.")
+
+        # Get Certificate for Withdrawal epoch 0 and verify it
+        we0_certHash = mc_node.getrawmempool()[0]
+        print("Withdrawal epoch 0 certificate hash = " + we0_certHash)
+        we0_cert = mc_node.getrawcertificate(we0_certHash, 1)
+        assert_equal(self.sc_nodes_bootstrap_info.sidechain_id, we0_cert["cert"]["scid"], "Sidechain Id in certificate is wrong.")
+        assert_equal(0, we0_cert["cert"]["epochNumber"], "Sidechain epoch number in certificate is wrong.")
+        assert_equal(we0_end_mcblock_hash, we0_cert["cert"]["endEpochBlockHash"], "Sidechain endEpochBlockHash in certificate is wrong.")
+        assert_equal(0, we0_cert["cert"]["totalAmount"], "Sidechain total amount in certificate is wrong.")
+
+        # Generate MC block and verify that certificate is present
+        we1_2_mcblock_hash = mc_node.generate(1)[0]
+        assert_equal(0, mc_node.getmempoolinfo()["size"], "Certificate expected to be removed from MC node mempool.")
+        assert_equal(1, len(mc_node.getblock(we1_2_mcblock_hash)["tx"]), "MC block expected to contain 1 transaction.")
+        assert_equal(1, len(mc_node.getblock(we1_2_mcblock_hash)["cert"]), "MC block expected to contain 1 Certificate.")
+        assert_equal(we0_certHash, mc_node.getblock(we1_2_mcblock_hash)["cert"][0], "MC block expected to contain certificate.")
+
+        #Write test data - sidechain forward transfer transaction and block, containing it.
+        bt_block = mc_node.getblock(we1_2_mcblock_hash, False)
+        bt_transaction = mc_node.gettransaction(we0_cert["certid"])["hex"]
+        self.write_test_data(test_data_file, "Sidechain backward transfer MC block", bt_block)
+        self.write_test_data(test_data_file, "Sidechain backward transfer MC transaction", bt_transaction)
+
+
+        # Generate SC block and verify that certificate is synced back
```

Comment:
Why do we need the rest of code?
Honestly, I don't see any benefits by using this test at all. It's just a copy-paste of BT test with printing of some data.
So, we can just use the original test.
As for me, the idea of having python script is to help automatically regenerate **all data** that we need for `MC Block` / `MC Tx` / `MC Cert` Unit tests.
I suggest to delete this file.